vision/mnist/mlp.jl

```diff
@@ -1,43 +1,89 @@
-using Flux, Flux.Data.MNIST, Statistics
-using Flux: onehotbatch, onecold, crossentropy, throttle
+using Flux, Statistics
+using Flux.Data: DataLoader
+using Flux: onehotbatch, onecold, crossentropy, throttle, @epochs
 using Base.Iterators: repeated
+using Parameters: @with_kw
 using CUDAapi
-if has_cuda()
+using MLDatasets
+if has_cuda()		# Check if CUDA is available
     @info "CUDA is on"
-    import CuArrays
+    import CuArrays		# If CUDA is available, import CuArrays
     CuArrays.allowscalar(false)
 end
 
+@with_kw mutable struct Args
+    η::Float64 = 3e-4       # learning rate
+    batchsize::Int = 1024   # batch size
+    epochs::Int = 10        # number of epochs
+    device::Function = gpu  # set as gpu, if gpu available
+end
+
+function get_data(args)
+    # Loading Dataset	
+    xtrain, ytrain = MLDatasets.MNIST.traindata(Float32)
+    xtest, ytest = MLDatasets.MNIST.testdata(Float32)
+	
+    # Reshape Data for flatten the each image into linear array
+    xtrain = reshape(xtrain,:,size(xtrain,3))
+    xtest = reshape(xtest,:,size(xtest,3))
 
-# Classify MNIST digits with a simple multi-layer-perceptron
+    # One-hot-encode the labels
+    ytrain, ytest = onehotbatch(ytrain, 0:9), onehotbatch(ytest, 0:9)
 
-imgs = MNIST.images()
-# Stack images into one large batch
-X = hcat(float.(reshape.(imgs, :))...) |> gpu
+    # Batching
+    train_data = DataLoader(xtrain, ytrain, batchsize=args.batchsize, shuffle=true)
+    test_data = DataLoader(xtest, ytest, batchsize=args.batchsize,)
+
+    return train_data, test_data
+end
 
-labels = MNIST.labels()
-# One-hot-encode the labels
-Y = onehotbatch(labels, 0:9) |> gpu
+function model(; imgsize=(28,28,1), nclasses=10)
+    return Chain(
+ 	    Dense(prod(imgsize), 32, relu),
+            Dense(32, nclasses),
+  	    softmax)
```

Comment:
could you remove `softmax` for all models and use `logitcrossentropy` as a loss instead? this should be numerically more stable